File: jsNCAA/scripts/_last14.mjs
import fs from "fs";
const h = JSON.parse(fs.readFileSync("data/history.json", "utf8"));
const runs = (h.runs || []).filter(r => r.burnIn !== true);
const last14 = runs.slice(-14);

let sw = 0, sl = 0, tw = 0, tl = 0;
for (const r of last14) {
  console.log(`${r.date}: ${(r.games || []).length} games`);
  for (const g of r.games || []) {
    if (g.sResult === "WIN") sw++;
    else if (g.sResult === "LOSS") sl++;
    if (g.oResult === "WIN") tw++;
    else if (g.oResult === "LOSS") tl++;
  }
}
console.log(`\nLAST 14 DAYS:`);
console.log(`Spreads: ${sw}-${sl} (${(sw / (sw + sl) * 100).toFixed(1)}%)  ${sw + sl} picks`);
console.log(`Totals:  ${tw}-${tl} (${(tw / (tw + tl) * 100).toFixed(1)}%)  ${tw + tl} picks`);
console.log(`Combined: ${sw + tw}-${sl + tl} (${((sw + tw) / (sw + tw + sl + tl) * 100).toFixed(1)}%)`);

File: pyNCAA/scripts/_last14.py
#!/usr/bin/env python3
# scripts/_last14.py — Last 14 days record summary
import json

h = json.loads(open("data/history.json").read())
runs = [r for r in (h.get("runs") or []) if not r.get("burnIn")]
last14 = runs[-14:]

sw, sl, tw, tl = 0, 0, 0, 0
for r in last14:
    print(f"{r.get('date')}: {len(r.get('games', []))} games")
    for g in (r.get("games") or []):
        if g.get("sResult") == "WIN": sw += 1
        elif g.get("sResult") == "LOSS": sl += 1
        if g.get("oResult") == "WIN": tw += 1
        elif g.get("oResult") == "LOSS": tl += 1

print(f"\nLAST 14 DAYS:")
print(f"Spreads: {sw}-{sl} ({sw / (sw + sl) * 100:.1f}%)  {sw + sl} picks")
print(f"Totals:  {tw}-{tl} ({tw / (tw + tl) * 100:.1f}%)  {tw + tl} picks")
print(f"Combined: {sw + tw}-{sl + tl} ({(sw + tw) / (sw + tw + sl + tl) * 100:.1f}%)")
